rq/worker.py

```diff
@@ -418,11 +418,22 @@ def _set_connection(self, connection: Optional['Redis']) -> 'Redis':
         """
         if connection is None:
             connection = get_current_connection()
-        current_socket_timeout = connection.connection_pool.connection_kwargs.get("socket_timeout")
-        if current_socket_timeout is None:
-            timeout_config = {"socket_timeout": self.connection_timeout}
-            connection.connection_pool.connection_kwargs.update(timeout_config)
-        return connection
+        if connection.__class__.__name__ == 'Redis':
+            current_socket_timeout = connection.connection_pool.connection_kwargs.get("socket_timeout")
+            if current_socket_timeout is None:
+                timeout_config = {"socket_timeout": self.connection_timeout}
+                connection.connection_pool.connection_kwargs.update(timeout_config)
+            return connection
```

Comment:
We'd also need to test this so test coverage doesn't go down. An easy way to do this would be to move this logic to a separate `setup_connection()` and pass both `Redis` and `RedisCluster` instances into it so we can easily test this.